### PR TITLE
stm32l4-multi: Add program flash raw write and erase to API

### PR DESCRIPTION
**Pull request by agkaminski**
JIRA: ISC-194

## Description

Please note that only 8 byte aligned and 8 byte quants write are allowed. STM32L4 flash memory features ECC, so rewrites of these 8 byte quants are not possible - page needs to be erased for these data to be modified.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## How Has This Been Tested?

- [ ] Already covered by automatic testing.
- [ ] New test added: (add PR link here).
- [x] Tested by hand on: (stm32l4).

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing linter checks and tests passed.
- [x] My changes generate no new compilation warnings for any of the targets.

## Special treatment

- [ ] This PR needs additional PRs to work (list the PRs, preferably in merge-order).
- [ ] I will merge this PR by myself when appropriate.

### CODE REVIEW

File: multi/stm32l4-multi/flash.c [hunk 1]
```diff
@@ -317,6 +317,56 @@ void flash_getInfo(flashinfo_t *info)
...
+
+	if (program_isValidAddress(offset, size) == 0)
+		return 0;
+
+	mutexLock(flash_common.lock);
+	_program_unlock();
+
+	ret = _flash_wait();
+	if (ret != 0) {
+		return ret;
```
Comment (Darchiv) [inline multi/stm32l4-multi/flash.c:332]: Return, but flash_common.lock mutex is not unlocked and _program_lock is not called

File: multi/stm32l4-multi/flash.c [hunk 2]
```diff
@@ -317,6 +317,56 @@ void flash_getInfo(flashinfo_t *info)
...
+
+	if (program_isValidAddress(offset, FLASH_PAGE_SIZE) == 0)
+		return 0;
+
+	mutexLock(flash_common.lock);
+	_program_unlock();
+
+	ret = _flash_wait();
+	if (ret != 0) {
+		return ret;
```
Comment (Darchiv) [inline multi/stm32l4-multi/flash.c:357]: Return, but flash_common.lock mutex is not unlocked and _program_lock is not called

File: multi/stm32l4-multi/flash.c (hunk 1 above)
Comment (agkaminski) [reply multi/stm32l4-multi/flash.c:332]: It's a bug copied from old code, I'll fix it everywhere, thank you

Comment (agkaminski) [reply multi/stm32l4-multi/flash.c]: Fixed, thanks

**Review by Darchiv — APPROVED** [review]
LGTM

**Pull request merged**